src/scheduler.c

```diff
@@ -393,6 +400,35 @@ schedule_adc_event(struct output_event *ev, struct decoder *d) {
   return 1;
 }
 
+void schedule_event(struct output_event *ev) {
+  switch(ev->type) {
+    case IGNITION_EVENT:
+      if (ignition_cut() || !config.decoder.valid) {
```

Comment:
Since we check decode validity here, nix it from toplevel decoder update.

Reply:
In fact all these checks were there because decoder state might change at any time, but this is no longer true. It will never change in the course of this run, so eliminate all of these cut/validity checks, do it once at the top level.

Reply:
We still need the checks for the reschedule callbacks, so leave these in place.  Certainly don't need to be volatile though.